client/client-lib/src/lib.rs

```diff
@@ -83,17 +84,26 @@ pub struct PaymentParameters {
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct UserClientConfig {
-    pub client_config: ClientConfig,
-    pub gateway: LightningGateway,
-}
+pub struct UserClientConfig(pub ClientConfig);
```

Comment:
Now that `UserClientConfig` just wraps `ClientConfig` do we still have a reason for this new-type ?
Or could we just do `type UserClient = Client<ClientConfig>` instead of `..Client<UserClientConfig>`

Reply:
Yea we could do that now. Will give it a try.

Reply:
Iirc we discussed that and came to the conclusion that the newtypes let us expose certain functionalities for e.g. only the user client, which might avoid certain kinds of user errors (trying to pay via a GW with a GW, potentially leading to a cyclic dependency).

Reply:
Ahh that's right